api/src/main/java/org/commonjava/indy/CustomJsonLayout.java

```diff
@@ -13,49 +13,62 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package org.commonjava.indy.bind.jaxrs;
+package org.commonjava.indy;
 
 import ch.qos.logback.classic.spi.ILoggingEvent;
 import ch.qos.logback.contrib.json.classic.JsonLayout;
-import com.fasterxml.jackson.core.JsonProcessingException;
-import org.commonjava.indy.conf.EnvironmentConfig;
-import org.commonjava.indy.model.core.io.IndyObjectMapper;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import javax.enterprise.inject.spi.CDI;
+import java.util.HashMap;
 import java.util.Map;
-
-import static org.commonjava.indy.bind.jaxrs.RequestContextConstants.ENVIRONMENT;
+import java.util.stream.Stream;
 
 /**
  * Created by yma on 2019/3/26.
  */
 public class CustomJsonLayout
         extends JsonLayout
 {
+    public static final String ENVIRONMENT = "environment";
+
     private final Logger logger = LoggerFactory.getLogger( getClass() );
 
+    private String environmentMappings;
+
+    private Map<String, String> envars;
+
+    public String getEnvironmentMappings()
+    {
+        return environmentMappings;
+    }
+
+    public void setEnvironmentMappings( final String environmentMappings )
+    {
+        this.environmentMappings = environmentMappings;
+
+        String[] mappings = environmentMappings == null ? new String[0] : environmentMappings.split( "\\s*,\\s*" );
+        envars = new HashMap<>();
+        Stream.of(mappings).forEach( kv ->{
+            String[] keyAlias = kv.split( "\\s*=\\s*" );
+            if ( keyAlias.length > 1 )
+            {
+                String value = System.getenv( keyAlias[0].trim() );
+                if ( value == null )
```

Comment:
nitpick. Maybe better to use StringUtils.isEmpty(value) ?

Reply:
That's a good point, because new `String("foo=").split("\\s*=\\s*")` would probably leave an array with two elements, and the last element would be an empty string, not null. I've adjusted it.